test/k8sT/fqdn.go

```diff
@@ -0,0 +1,168 @@
+
+		err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+		Expect(err).Should(BeNil(), "Demo app is not ready after timeout")
+
+		appPods = helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
+
+		err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=bind", 300)
+		Expect(err).Should(BeNil(), "Demo app is not ready after timeout")
+
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport(helpers.KubeSystemNamespace,
+			"cilium service list",
+			"cilium endpoint list")
+	})
+
+	AfterAll(func() {
+		_ = kubectl.Delete(bindManifest)
+		_ = kubectl.Delete(demoManifest)
+		ExpectAllPodsTerminated(kubectl)
+	})
+
+	JustBeforeEach(func() {
+		microscopeErr, microscopeCancel = kubectl.MicroscopeStart()
+		Expect(microscopeErr).To(BeNil(), "Microscope cannot be started")
+		backgroundCancel, backgroundError = kubectl.BackgroundReport("uptime")
+		Expect(backgroundError).To(BeNil(), "Cannot start background report process")
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+		Expect(microscopeCancel()).To(BeNil(), "cannot stop microscope")
+		backgroundCancel()
+	})
+
+	AfterEach(func() {
+		_ = kubectl.Exec(fmt.Sprintf("%s delete --all cnp", helpers.KubectlCmd))
+	})
+
+	It("Restart Cilium validate that FQDN is still working", func() {
+		connectivityTest := func() {
+			res := kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(worldTarget))
+			ExpectWithOffset(1, res).To(helpers.CMDSuccess(), "%q cannot curl to %q",
+				appPods[helpers.App2], worldTarget)
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(worldInvalidTarget))
+			ExpectWithOffset(1, res).ShouldNot(helpers.CMDSuccess(),
+				"%q can curl to %q when it should fail", appPods[helpers.App2], worldInvalidTarget)
+
+			By("Testing that connection using IP is working correctly")
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(worldTargetIP))
+			res.ExpectSuccess("%q cannot curl to %q during restart", helpers.App2, worldTargetIP)
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(worldInvalidTargetIP))
+			res.ExpectFail("%q can  connect when it should not work", helpers.App2)
+		}
+
+		fqndProxyPolicy := helpers.ManifestGet("fqdn-proxy-policy.yaml")
+
+		_, err := kubectl.CiliumPolicyAction(
+			helpers.KubeSystemNamespace, fqndProxyPolicy,
+			helpers.KubectlApply, helpers.HelperTimeout)
+		Expect(err).To(BeNil(), "Cannot install fqdn proxy policy")
+
+		connectivityTest()
+		By("Deleting cilium pods")
+
+		res := kubectl.Exec(fmt.Sprintf("%s -n %s delete pods -l k8s-app=cilium",
+			helpers.KubectlCmd, helpers.KubeSystemNamespace))
+		res.ExpectSuccess()
+
+		By("Testing connectivity when cilium is restoring using IPS without DNS")
```

Comment:
To confirm - this works because the IPs themselves are mapped to an identity which is stored in the policy map for the endpoint, which is persistent across restarts?

Reply:
yes, @raybejjani made the work this week.